packages/x-scheduler/src/primitives/use-event-occurrences-with-timeline-position/useEventOccurrencesWithTimelinePosition.ts

```diff
@@ -3,6 +3,58 @@ import { CalendarEventOccurrence, CalendarEventOccurrenceWithTimePosition } from
 import { useAdapter } from '../utils/adapter/useAdapter';
 import { Adapter } from '../utils/adapter/types';
 
+const innerOccurrencesWithTimelinePosition = (
+  adapter: Adapter,
+  occurrences: CalendarEventOccurrence[],
+  maxColumnSpan: number,
+) => {
+  const conflicts = buildOccurrenceConflicts(adapter, occurrences);
+
+  const { firstIndexLookup, maxIndex } = buildFirstIndexLookup(conflicts);
+
+  const lastIndexLookup = buildLastIndexLookup(
+    conflicts,
+    firstIndexLookup,
+    maxIndex,
+    maxColumnSpan,
+  );
+
+  const occurrencesWithPosition = occurrences.map((occurrence) => ({
+    ...occurrence,
+    position: {
+      firstIndex: firstIndexLookup[occurrence.key],
+      lastIndex: lastIndexLookup[occurrence.key],
+    },
+  }));
+
+  return { occurrences: occurrencesWithPosition, maxIndex };
+};
+
+export function useEventOccurrencesWithTimelinePositionForMap(
```

Comment:
I think you can re-use `useEventOccurreceWithTimelinePosition`, if you place it inside the component that renders a single resource.

It's probably better, because when we will have drag and drop, we will want a component per resource so that we only re-render the impacted component when dragging an event.